### ISSUE TITLE
Touchpad Gesture options

### ISSUE THREAD

**Issue by pknepps** [pop-os/cosmic-settings#697]
The following is shown, but incomplete under the touchpad settings page:

![image](https://github.com/user-attachments/assets/94c8a92c-30f7-4770-b1be-d83fe96aa1e3)

Under the design document, the touchpad page shows the following: 
![image](https://github.com/user-attachments/assets/2502996c-8578-4241-a87e-de474617c259)

These settings need to be added, and be implemented through cosmic-comp.

Tied to cosmic-comp issue pop-os/cosmic-comp#315

I would like to begin working on this unless a maintainer says otherwise.

**Comment by nebrelbug** [comment]
I'm excited for this! It's the only thing missing from my old GNOME + VShell + Pop!_Shell workflow.

**Comment by pknepps** [comment]
I have become extremely busy with work for the past few months and have abandoned this for now. If someone else wants to give it go, it is free.

**Comment by Etienne-TR** [comment]
Gestures are really very important, I hope someone will know how to implement them!

The best would be to be able to configure the actions for each gesture.

**Comment by crazy-vedic** [comment]
I would also be interested in such a feature.

**Comment by lveillard** [comment]
it would be lovely to be able to customize three finger gestures granularly like in windows 11.

in my case
- 3 fingers tap is play/pause
- 3 up = next song
- 3 down = previous song
- 3 right = previos navigation (works in file explorers not only web browsers)
- 3 left = forward navigation

And a gesture not available in windows 11 which is amazing in macOs is the fast 2 fingers to the right / left which are the backwards/forwards navigation gestures and were the only gestures i missed from mac in windows, as windows only captures nornal 2 finger gestures and it's not possible to configure "fast" ones

btw that "fast" gesture does work natively in chrome!

**Comment by andmog** [comment]
I agree. The 2-finger gesture to go back and forth in a web browser and folders (Finder), like in Mac OS X are amazing.
Also, the 2-finger gesture to zoom in and out in the web browser and PDF files.
I would need them in Cosmic

**Comment by nadav-db** [comment]
> I agree. The 2-finger gesture to go back and forth in a web browser and folders (Finder), like in Mac OS X are amazing.
> Also, the 2-finger gesture to zoom in and out in the web browser and PDF files.
> I would need them in Cosmic

Just so you'll know, overscroll history navigation is available in chrome browser (I think Firefox supports it too, but I use chrome)
You just need to add a flag to the exec command in the desktop entry. If you still need it I can find it for you on my computer

**Comment by WatchMkr** [comment]
Defaults as discussed at the Poland Summit:
- Four finger
  - Workspaces
- Three finger
  - Workspace overview
  - App library
- Two finger
  - Pinch/spread zoom in app
  - With modifier for desktop zoom (Super?)
    - Nearest neighbor accessibility zooming option - perhaps default

**Comment by mmstick** [comment]
Custom touchpad gestures implemented in https://github.com/pop-os/cosmic-settings-daemon/pull/63 and https://github.com/pop-os/cosmic-comp/pull/396, but requires to be rebased and updated.
